src/quest_manager/models.py

```diff
@@ -47,6 +47,10 @@ def get_icon_url(self):
         else:
             return SiteConfig.get().get_default_icon_url()
 
+    def quest_count(self):
+        """ Returns the total number of quests available in this campaign."""
+        return self.quest_set.all().visible().not_archived().count()
```

Comment:
The queryset you are using is identical to the one below.  To keep these to the same and make the code more DRY, could you add a new method `current_quests()` and have it return this  queryset: `self.quest_set.all().visible().not_archived()`

Then, update both of these methods (quest_count and xp_sum) so that they use it: `self.current_quests().count()` and `self.current_quests().aggregate(Sum('xp'))['xp__sum']`

That way in the future if the way we figure out the quests we want, it will affect both equally.

Also add test for the new method.

Reply:
```
    def test_current_quests(self):
        """ Test that the queryset of all quests in a campaign is returned correctly """

        # assert that the current campaign has no quests
        self.assertEqual(self.category.current_quests().count(), 0)

        # create some quests as part of the test campaign, some are invalid and won't be included
        baker.make(Quest, campaign=self.category, xp=0)  # included in queryset
        baker.make(Quest, campaign=self.category, xp=0)  # included in queryset
        baker.make(Quest, visible_to_students=False, xp=8)  # NOT included in queryset because not visible
        baker.make(Quest, archived=True, xp=16)  # NOT included in queryset because archived
        baker.make(Quest, xp=4)  # NOT included in queryset because in a different campaign

        # assert that the current campaign has 2 valid quests after additions
        self.assertEqual(self.category.current_quests().count(), 2)
```
method + test added